### PR TITLE
Update build file and a bunch of deps. Travis->GitHubActions

### PR DESCRIPTION
**Pull request by edenman**
New:
- Gradle
- AGP
- compileSdk 31
- Java 11
- annotations 1.3
- junit
- robolectric
- GitHub Actions config

Removed:
- Old busted Maven publish stuff (I'll fix this in the next PR)
- Travis config

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - uses: gradle/wrapper-validation-action@v1.0.4
+    - name: set up JDK
+      uses: actions/setup-java@v2
+      with:
+        java-version: 11
+        distribution: 'temurin'
```
Comment (jrodbx) [inline .github/workflows/android.yml:21]: not sure what the diff between temurin and zulu is.  I see zulu in a few places.

File: build.gradle [hunk 2]
```diff
@@ -1,61 +1,44 @@
...
     mavenCentral()
     maven {
-      url 'https://maven.google.com/'
-      name 'Google'
+      url "https://plugins.gradle.org/m2/"
     }
   }
   dependencies {
-    classpath 'com.android.tools.build:gradle:1.5.0'
+    classpath 'com.android.tools.build:gradle:7.3.0-alpha07'
```
Comment (jrodbx) [inline build.gradle:10]: cutting edge

File: gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
-#Sat Feb 06 10:49:01 CST 2016
+#Mon Mar 07 13:40:17 NZDT 2022
 distributionBase=GRADLE_USER_HOME
+distributionUrl=https\://services.gradle.org/distributions/gradle-7.4-bin.zip
```
Comment (jrodbx) [inline gradle/wrapper/gradle-wrapper.properties:3]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-7.4.2-bin.zip
```

File: build.gradle [hunk 4]
```diff
@@ -81,12 +64,3 @@ task checkstyle(type: Checkstyle) {
...
 }
-
-repositories {
-  maven {
-    url 'https://maven.google.com/'
-    name 'Google'
-  }
-}
-
-apply from: 'gradle-mvn-push.gradle'
```
Comment (jrodbx) [inline build.gradle:92]: if you're deleting this, ya gonna need vanniktech plugin or something else to publish

File: .github/workflows/android.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
```
Comment (jrodbx) [inline .github/workflows/android.yml:11]: nit:
```suggestion
```

**Review by jrodbx — APPROVED** [review]

File: build.gradle (hunk 4 above)
Comment (jrodbx) [reply build.gradle:92]: > Old busted Maven publish stuff (I'll fix this in the next PR)

d'oh!

File: .github/workflows/android.yml (hunk 1 above)
Comment (edenman) [reply .github/workflows/android.yml:21]: yeah, i was trying them all when trying to get a test to work in android-times-square.  i'll switch to zulu, shouldn't matter here

**Pull request merged**